File: app/src/main/java/edu/hm/cs/krivoj/mensapp/api/price/Table.java
package edu.hm.cs.krivoj.mensapp.api.price;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public class Table implements Serializable {

  private Map<String, Category> map = new HashMap<>();

  public Category put(String key, Category value) {
    return map.put(key, value);
  }

  private static String cleanKey(String key) {
    return key.replaceAll("(Aktionsgericht)|(Aktionsessen)|(Biogericht)|(Bioessen)",
        "Bio-/Aktionsgericht").trim();
  }

  public boolean contains(String key) {
    return map.containsKey(cleanKey(key));
  }

  public String getStudentPrice(String key) {
    if (!contains(key)) {
      return "";
    }
    String clean = cleanKey(key);
    return map.get(clean).getStudent();
  }

  public String getEmployeePrice(String key) {
    if (!contains(key)) {
      return "";
    }
    String clean = cleanKey(key);
    return map.get(clean).getEmployee();
  }

  public String getGuestPrice(String key) {
    if (!contains(key)) {
      return "";
    }
    String clean = cleanKey(key);
    return map.get(clean).getGuest();
  }
}

File: app/src/main/java/edu/hm/cs/krivoj/mensapp/api/price/Table.kt
package edu.hm.cs.krivoj.mensapp.api.price

import java.io.Serializable
import java.util.*

class Table : Serializable {

    private val map = HashMap<String, Category?>()

    fun put(key: String, value: Category): Category? =  map.put(key, value)

    private fun cleanKey(key: String): String = key.replace("(Aktionsgericht)|(Aktionsessen)|(Biogericht)|(Bioessen)".toRegex(), "Bio-/Aktionsgericht").trim { it <= ' ' }

    operator fun contains(key: String): Boolean = map.containsKey(cleanKey(key))

    fun getStudentPrice(key: String): String {
        if (!contains(key)) {
            return ""
        }
        val clean = cleanKey(key)
        return map[clean]!!.student
    }

    fun getEmployeePrice(key: String): String {
        if (!contains(key)) {
            return ""
        }
        val clean = cleanKey(key)
        return map[clean]!!.employee
    }

    fun getGuestPrice(key: String): String {
        if (!contains(key)) {
            return ""
        }
        val clean = cleanKey(key)
        return map[clean]!!.guest
    }
}
